brage-import/src/test/java/no/sikt/nva/brage/migration/lambda/BrageEntryEventConsumerTest.java

```diff
@@ -930,6 +939,88 @@ void shouldSavePublicationBeforeUpdateInS3() throws IOException, BadRequestExcep
+        var doi = randomDoi();
+        var existingPublicationWithoutCristinIdentifier = persistPublicationWithCristinIdAndDoi(null, doi);
+        assertThat(existingPublicationWithoutCristinIdentifier.getHandle(), is(nullValue()));
+
+        var listOfExistingPublications = List.of(existingPublicationWithoutCristinIdentifier);
+        var searchResourceApiResponseSingleHit = new SearchResourceApiResponse(listOfExistingPublications.size(),
+                                                                               listOfExistingPublications);
+        var singleHitOptional = Optional.of(searchResourceApiResponseSingleHit.toString());
+        when(uriRetriever.getRawContent(any(), any())).thenReturn(singleHitOptional);
+
+        var nvaBrageMigrationDataGenerator = new NvaBrageMigrationDataGenerator.Builder()
+                                                 .withPublishedDate(null)
+                                                 .withType(TYPE_BOOK)
+                                                 .withDoi(doi)
+                                                 .build();
+
+        var brageRecord = nvaBrageMigrationDataGenerator.getBrageRecord();
+        var s3Event = createNewBrageRecordEvent(brageRecord);
+        var publication = handler.handleRequest(s3Event, CONTEXT);
+        assertThat(publication.getHandle(), is(equalTo(brageRecord.getId())));
+        assertThat(publication.getIdentifier(),
+                   is(equalTo(existingPublicationWithoutCristinIdentifier.getIdentifier())));
+
+        var actualStoredHandleString =
+            extractActualHandleReportFromS3Client(s3Event, existingPublicationWithoutCristinIdentifier);
+        assertThat(actualStoredHandleString,
+                   is(equalTo(nvaBrageMigrationDataGenerator.getBrageRecord().getId().toString())));
+    }
+
+    @Test
+    void shouldNotMergePublicationsIfNoDoiDuplicateExists() throws IOException {
+        var searchResourceApiResponseNoHits = new SearchResourceApiResponse(0, null);
+        var noHitsOptional = Optional.of(searchResourceApiResponseNoHits.toString());
+        when(uriRetriever.getRawContent(any(), any())).thenReturn(noHitsOptional);
+
+        var nvaBrageMigrationDataGenerator = new NvaBrageMigrationDataGenerator.Builder()
+                                                 .withPublishedDate(null)
+                                                 .withType(TYPE_BOOK)
+                                                 .withDoi(randomDoi())
+                                                 .build();
+
+        var brageRecord = nvaBrageMigrationDataGenerator.getBrageRecord();
+        var s3Event = createNewBrageRecordEvent(brageRecord);
+        var publication = handler.handleRequest(s3Event, CONTEXT);
+        assertThat(publication.getHandle(), is(equalTo(brageRecord.getId())));
+
+        var actualStoredHandleString = extractActualHandleReportFromS3Client(s3Event, publication);
```

Comment:
lurer på om du kan hente ut forventet NVA publikasjon fra brageGeneratoren.

Reply:
.getNvaPublication() fra brageGeneratoren mangler en del felter som gjør at testen kaster NullPointerExceptions. Mangler bl. a. identifier. Funker om man ikke går via extractActualHandleReportFromS3Client(), og heller går direkte på brageGeneratorens nvaPublication for å hente ut handler-string. Var det det du tenkte på?

Reply:
Da lar vi det stå slik som det er nå. Trodde det skulle være mulig å bare hente den derfra.